pkg/login/social/generic_oauth_test.go

```diff
@@ -427,3 +427,66 @@ func TestUserInfoSearchesForLogin(t *testing.T) {
 		}
 	})
 }
+
+func TestPayloadCompression(t *testing.T) {
+	provider := SocialGenericOAuth{
+		SocialBase: &SocialBase{
+			log: log.NewWithLevel("generic_oauth_test", log15.LvlDebug),
+		},
+		emailAttributePath: "email",
+	}
+
+	tests := []struct {
+		Name          string
+		OAuth2Extra   interface{}
+		ExpectResult  bool
+		ExpectedEmail string
```

Comment:
I still think it's better to remove `ExpectResult`, since `ExpectResult: true` and `ExpectedEmail: ""` would be an invalid combination, right? As such, you might as well avoid the tedium of the extra boolean attribute.

Reply:
I'm fine either way.  I just updated the test and pushed to the PR branch.